jdk.sh

```diff
@@ -0,0 +1,32 @@
+package: JDK
+version: "10.0.1"
+build_requires:
+ - curl
+---
+#!/bin/bash -e
+ZIP_NAME="jdk-$PKGVERSION_linux-x64_bin.tar.gz"
+UNZIP_NAME="jdk-$PKGVERSION"
+URL="http://download.oracle.com/otn-pub/java/jdk/10.0.1+10/fb4372174a714e6b8c52526dc134031e/$ZIP_NAME"
+curl -b "oraclelicense=a" -L -O $URL
+tar xvfz $ZIP_NAME
+rsync -av $UNZIP_NAME/ $INSTALLROOT/
+export  PATH=$INSTALLROOT/Java/bin:$PATH
```

Comment:
Very good, is it possible to download & unpack directly in the installation directory, without an intermediate tar file and an intermediate unpack directory?

I have tested the following one-liner* (*it's one line plus some boilerplate) and it works:

```bash
mkdir -p "$INSTALLROOT"
curl -b 'oraclelicense=a' -L http://download.oracle.com/otn-pub/java/jdk/10.0.1+10/fb4372174a714e6b8c52526dc134031e/jdk-10.0.1_linux-x64_bin.tar.gz | tar --strip-components 1 -C "$INSTALLROOT" -xvvf -
```

Also, the `export PATH` part is not needed (plus I believe the exported path is wrong).

This recipe is also missing:

* A first line breaking if we are on macOS (`[[ $(uname) != Linux ]] && { echo "Works on Linux only"; false; }`)
* A `prefer_system` to pick the JDK from the system if appropriate (you should check for `javac` I believe, _without using `which`_).

Reply:
I'd leave PATH in so we don't have to guess where `java` can be found

Reply:
The PATH in this case is in the build environment.
I think @dberzano is correct that it is not needed.
You have it set in the runtime environment module:
```
set JAVA_ROOT \$::env(BASEDIR)/$PKGNAME/\$version
prepend-path PATH \$JAVA_ROOT/bin
```